spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -2344,7 +2344,7 @@ public <T> T invoke(OperationsCallback<T> action, @Nullable com.rabbitmq.client.
 				if (channel == null) {
 					throw new IllegalStateException("Connection returned a null channel");
 				}
-				if (!connectionFactory.isPublisherConfirms()) {
+				if (!connectionFactory.isPublisherConfirms() && !connectionFactory.isSimplePublisherConfirms()) {
```

Comment:
Isn't the change in #2651 more precise and accurate?

Reply:
Sorry, this was still a bit WIP. Added this change here as well and added also test for this. Hope this covers this completely.